pkg/controller/instance/instance_controller.go

```diff
@@ -429,24 +408,23 @@ func PipesMap(planName string, plan *v1beta1.Plan, tasks []v1beta1.Task, emeta *
 	return pipes, nil
 }
 
-// startPlanExecution mark plan as to be executed
-// this modifies the instance.Status as well as instance.Metadata.Annotation (to save snapshot if needed)
-func startPlanExecution(i *v1beta1.Instance, planName string, ov *v1beta1.OperatorVersion) error {
-	if i.NoPlanEverExecuted() || isUpgradePlan(planName) {
-		ensurePlanStatusInitialized(i, ov)
+// resetPlanStatusIfThePlanIsNew method resets a PlanStatus for a passed plan name and instance *IF* this is a newly
+// scheduled plan (UID has changed). In this case Plan/phase/step statuses are set to ExecutionPending meaning that the
+// controller will restart plan execution. Otherwise (the plan is old), nothing is changed.
+func resetPlanStatusIfThePlanIsNew(i *v1beta1.Instance, plan string, uid types.UID) (*v1beta1.PlanStatus, error) {
```

Comment:
Nit: I don't think we have "The" in any of the funcs

```suggestion
func resetPlanStatusIfPlanIsNew(i *v1beta1.Instance, plan string, uid types.UID) (*v1beta1.PlanStatus, error) {
```